openshift-kube-apiserver/admission/customresourcevalidation/route/validation_wrapper_test.go

```diff
@@ -0,0 +1,41 @@
+package route
+
+import (
+	"fmt"
+	"testing"
+
+	"k8s.io/client-go/kubernetes/fake"
+)
+
+// setter is only available in unit-tests
+func (a *validateCustomResourceWithClient) setupWithFakeClient() {
+	c := fake.NewSimpleClientset()
+	a.secretsGetter = c.CoreV1()
+	a.sarGetter = c.AuthorizationV1()
+	a.routeValidationOptsGetter = NewRouteValidationOpts()
+}
+
+// TestRouteValidateInitialization will check if validator client is
+// conformant with admission.InitializationValidator interface
+func TestRouteValidateInitialization(t *testing.T) {
+	admitter, err := NewValidateRoute()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	routeAdmitter, ok := admitter.(*validateCustomResourceWithClient)
+	if !ok {
+		t.Fatal(fmt.Errorf(
+			"could not type cast returned value of NewValidateRoute() into type validateCustomResourceWithClient, " +
+				"perhaps you changed the type in the implementation but not in the tests!"))
+	}
+
+	// unit test specific logic as a replacement for routeAdmitter.SetRESTClientConfig(...)
+	routeAdmitter.setupWithFakeClient()
+
+	// admission.InitializationValidator -> ValidateInitialization()
+	err = routeAdmitter.ValidateInitialization()
+	if err != nil {
+		t.Fatal(err)
+	}
+}
```

Comment:
You could merge this test with `TestValidateRoutePlugin`—just add the call and error check for `ValidateInitialization` to `TestValidateRoutePlugin`, and you'll have exactly the same code coverage.  I don't mind if you prefer to have it as a separate test though.

Reply:
Makes sense, I've updated to merge the two into the same `TestValidateRoutePlugin` function.